libvast/native-plugins/summarize.cpp

```diff
@@ -735,6 +735,52 @@ class summarize_operator : public pipeline_operator {
   std::unordered_map<type, std::vector<pipeline_batch>> blacklist_ = {};
 };
 
+caf::expected<std::unique_ptr<pipeline_operator>>
+make_summarize_operator(const record& config) {
+  auto parsed_config = configuration::make(config);
+  if (!parsed_config)
+    return parsed_config.error();
+  return std::make_unique<summarize_operator>(std::move(*parsed_config));
+}
+
+std::vector<std::string>
+get_keys_from_legacy_sections(const data::variant& legacy_section) {
```

Comment:
```suggestion
get_keys_from_legacy_sections(const data& legacy_section) {
```

`data::variant` is an implementation detail that should not be needed here.